tabcmd/commands/datasources_and_workbooks/export_command.py

```diff
@@ -73,65 +56,103 @@ def run_command(args):
 
-        try:
+        print(args.pagelayout, args.pagesize, args.filename, args.width, args.height, args.filter)
 
+        try:
             if args.fullpdf:  # it's a workbook
                 workbook_item = ExportCommand.get_wb_by_content_url(logger, server, wb_content_url)
-                output = ExportCommand.download_wb_pdf(server, workbook_item)
+                output = ExportCommand.download_wb_pdf(server, workbook_item, args.url, logger)
                 default_filename = "{}.pdf".format(workbook_item.name)
 
             elif args.pdf or args.png or args.csv:  # it's a view
                 view_item = ExportCommand.get_view_by_content_url(logger, server, view_content_url)
 
                 if args.pdf:
-                    output = ExportCommand.download_view_pdf(server, view_item)
+                    output = ExportCommand.download_view_pdf(server, view_item, args.url, logger)
                     default_filename = "{}.pdf".format(view_item.name)
                 elif args.csv:
-                    output = ExportCommand.download_csv(server, view_item)
+                    output = ExportCommand.download_csv(server, view_item, args.url, logger)
                     default_filename = "{}.csv".format(view_item.name)
                 elif args.png:
-                    output = ExportCommand.download_png(server, view_item)
+                    output = ExportCommand.download_png(server, view_item, args.url, logger)
                     default_filename = "{}.png".format(view_item.name)
 
         except Exception as e:
             Errors.exit_with_error(logger, _("publish.errors.unexpected_server_response").format(""), e)
 
         try:
             save_name = args.filename or default_filename
-            ExportCommand.save_to_file(logger, output, save_name)
+            if args.csv:
+                ExportCommand.save_to_data_file(logger, output, save_name)
+            else:
+                ExportCommand.save_to_file(logger, output, save_name)
 
         except Exception as e:
             Errors.exit_with_error(logger, "Error saving to file", e)
 
     @staticmethod
-    def download_wb_pdf(server, workbook_item):
+    def extract_query_params(request_options: TSC.PDFRequestOptions, url, logger=None):
+        try:
+            # todo make logging better
+            logger = logger or log(ExportCommand.__class__.__name__, "DEBUG")
+            logger.debug(url)
+
+            if "?" in url:
+                query = url.split("?")[1]
+            else:
+                logger.trace("No params?")
+                return None
+            params = query.split("&")
+            logger.trace(params)
+            for value in params:
+                filter = value.split("=")
+                request_options.vf(filter[0], filter[1])
+        except BaseException as e:
+            logger.error("Error building filter params", e)
+            ExportCommand.log_stack(logger)  # type: ignore
+        return request_options
```

Comment:
Do we even need to return `request_options` here? I'm assuming the original `request_options` argument passed in will have its `view_filters` field updated.

Reply:
good call, we do not. Fixed.

Reply:
Was this one still pending?